### PR TITLE
[HMS Environment] CANDLEPIN-694: Add environment types to database and API model

### PR DESCRIPTION
**Pull request by Januson**

### CODE REVIEW

File: spec-tests/src/test/java/org/candlepin/spec/environments/EnvironmentSpecTest.java [hunk 1]
```diff
@@ -144,6 +144,18 @@ public void shouldFindEnvironmentByName() {
             .returns(env.getId(), EnvironmentDTO::getId);
     }
 
+    @Test
+    public void shouldLoverCaseEnvironmentType() {
```
Comment (JoshAlbrecht23) [inline spec-tests/src/test/java/org/candlepin/spec/environments/EnvironmentSpecTest.java:148]: Typo: `shouldLoverCaseEnvironmentType` -> `shouldLowerCaseEnvironmentType`

**Review by JoshAlbrecht23 — CHANGES_REQUESTED** [review]
Looks good, I just noticed a small typo in a test name!

File: spec-tests/src/test/java/org/candlepin/spec/environments/EnvironmentSpecTest.java (hunk 1 above)
Comment (Januson) [reply spec-tests/src/test/java/org/candlepin/spec/environments/EnvironmentSpecTest.java]: Done.

**Review by JoshAlbrecht23 — APPROVED** [review]

File: src/test/java/org/candlepin/dto/api/v1/EnvironmentTranslatorTest.java [hunk 2]
```diff
@@ -66,6 +66,7 @@ protected Environment initSourceObject() {
 
         source.setId("test_id");
         source.setName("test_name");
+        source.setName("custom");
```
Comment (Ceiu) [inline src/test/java/org/candlepin/dto/api/v1/EnvironmentTranslatorTest.java:69]: I'm guessing this is intended to be `setType`?

**Review by Ceiu — CHANGES_REQUESTED** [review]

File: src/test/java/org/candlepin/dto/api/v1/EnvironmentTranslatorTest.java (hunk 2 above)
Comment (Januson) [reply src/test/java/org/candlepin/dto/api/v1/EnvironmentTranslatorTest.java]: Yes. Done.

**Review by Ceiu — APPROVED** [review]

**Pull request merged**